components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandlerTest.java

```diff
@@ -98,4 +140,58 @@ public void testResolvePassword() throws Exception {
     }
+
+    @Test
+    public void testHandleWithV2RolesOverrideAttributes() throws Exception {
+
+        List<String> roleIdList = new ArrayList<>();
+        roleIdList.add("role1");
+        roleIdList.add("role2");
+        String subject = "testUser";
+        String userId = "12231321";
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(FrameworkConstants.IDP_ID, "testIdp");
+        attributes.put(FrameworkConstants.ASSOCIATED_ID, "testUser");
+        attributes.put(FrameworkConstants.USER_ID_CLAIM, "12231321");
+
+        String provisioningUserStoreId = IdentityApplicationConstants.AS_IN_USERNAME_USERSTORE_FOR_JIT;
+        String tenantDomain = "carbon.super";
+
+        when(mockRealmService.getTenantManager()).thenReturn(mockTenantManager);
+        when(mockTenantManager.getTenantId(tenantDomain)).thenReturn(-1234);
+        when(mockRealmService.getTenantUserRealm(-1234)).thenReturn(mockUserRealm);
+        when(mockUserRealm.getUserStoreManager()).thenReturn(mockUserStoreManager);
+        when(mockUserStoreManager.getSecondaryUserStoreManager("PRIMARY")).thenReturn(mockUserStoreManager);
+        when(mockUserStoreManager.getRealmConfiguration()).thenReturn(mockRealmConfiguration);
+        when(mockUserRealm.getRealmConfiguration()).thenReturn(mockRealmConfiguration);
+        when(mockRealmConfiguration.isPrimary()).thenReturn(true);
+        when(mockRealmConfiguration.getAdminUserName()).thenReturn("admin");
+        when(mockRealmConfiguration.getEveryOneRoleName()).thenReturn("everyone");
+
+        when(mockUserStoreManager.isExistingUser(subject)).thenReturn(true);
+        when(mockRealmConfiguration.isPrimary()).thenReturn(true);
+
+        when(mockUserStoreManager.isExistingUser(subject)).thenReturn(true);
+
+        Map<String, Object> threadLocalProperties = new HashMap<>();
+        threadLocalProperties.put(FrameworkConstants.ATTRIBUTE_SYNC_METHOD, FrameworkConstants.OVERRIDE_ALL);
+        IdentityUtil.threadLocalProperties.set(threadLocalProperties);
+
+        when(mockFederatedAssociationManager.getUserForFederatedAssociation(tenantDomain, "testIdp", subject))
+                .thenReturn(subject);
+        when(mockOrganizationManager.resolveOrganizationId(tenantDomain)).thenReturn(SUPER_ORG_ID);
+        when(mockRoleManagementService.getRoleIdListOfUser(userId, tenantDomain))
+                .thenReturn(roleIdList);
+
+        try (MockedStatic<FrameworkUtils> frameworkUtils = mockStatic(FrameworkUtils.class)) {
+
+            frameworkUtils.when(() -> FrameworkUtils.resolveUserIdFromUsername(mockUserStoreManager, subject))
+                    .thenReturn(userId);
+            frameworkUtils.when(FrameworkUtils::getFederatedAssociationManager)
+                    .thenReturn(mockFederatedAssociationManager);
+            provisioningHandler.handleWithV2Roles(roleIdList, subject, attributes, provisioningUserStoreId,
```

Comment:
Don't we need to assert whether the user-id is not changed here?

Reply:
The user store manager is a mocked class, hence actual user claim update doesn't happen. The unit tests were added as the entire flow is not covered by any test case.

Reply:
Ack